govcd/vapptemplate.go

```diff
@@ -123,3 +123,16 @@ func (vAppTemplate *VAppTemplate) UpdateAsync() (Task, error) {
 	return vAppTemplate.client.ExecuteTaskRequest(url, http.MethodPut,
 		types.MimeVAppTemplate, "error updating vApp template item: %s", vappTemplatePayload)
 }
+
+// Delete deletes the VAppTemplate Item, returning an error if the VCD call fails.
```

Comment:
Where does this method act? 
Does it delete the vApp template from a catalog?
How is this function different from CatalogItem.Delete?

Reply:
The result and behaviour would be indeed equivalent to `CatalogItem.Delete`, the only difference is the implementation as the endpoint and receiver object is the vApp Template entity.